client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx

```diff
@@ -28,10 +28,10 @@ export default function ThankYouJetpackSearchProduct( {
 		getJetpackSearchDashboardUrl( state, siteId as number )
 	);
 
-	const productButtonLabel = selectedSite?.jetpack
```

Comment:
@wongasy I definitely agree that the site should always be defined, but TypeScript sees `selectedSite` as potentially `null` or `undefined` and throws an error here. I opted for conditional chains to suppress that, but we could also go another route if you prefer? Maybe an `if ( selectedSite ) ...` check? That would lead to both variables being `undefined` if the site somehow wasn't truthy, but that shouldn't happen at this point anyway, I don't think. Curious what your thoughts on best approach 🙂

Reply:
Playing around with this a bit more, I think I'm still leaning towards the optional chaining to resolve the type error.

Thought process:
- both of these variables (`productButtonLabel` and `productButtonHref`) are being declared via a ternaries based on the truthiness of `siteId.jetpack`.
- with optional chaining, if `siteId` is somehow `null` or `undefined`, the ternary will evaluate as `undefined` and return a falsey result... which is exactly what we'd want to happen, since in this case `siteId.jetpack` is also automatically `undefined` and falsey. 
- Basically if `siteId` proves to be falsey, the behavior of this code will be the same as if the `jetpack` prop itself were falsey, which is actually what we're trying to evaluate in the first place.

I'm hoping my description here makes sense 😅 If it does, perhaps I just add a quick comment explaining the reasoning behind the optional chaining?

If not, happy to explore/learn a better way to tackle this one!

Reply:
Thanks for the explanation! I didn't realize the optional was meant to resolve the type error. I'll revert back to conditional chaining.